include/kernel.h

```diff
@@ -3022,6 +3022,26 @@ struct k_delayed_work {
 	struct k_work_q *work_q;
 };
 
+/**
+ * @brief Structure that pairs a work item with a timeout.
+ *
+ * Used with k_work_schedule_for_queue() to allow a work item to be
+ * automatically submitted after a delay.
+ *
+ * No fields in this structure are to be accessed directly.
+ */
+struct k_work_delayable {
+	/* Internal work item that is submitted. */
+	struct k_work work;
+
+	/* Timeout used for delayed submission. */
+	struct _timeout timeout;
+
+	/* Pointer to the work queue to which the item will be submitted. */
+	/* TBD: atomic_ptr_t? */
```

Comment:
any details on the use of atomic_ptr_t here?

Reply:
Thanks, that comment should be removed.  I had some thought of using null/non-null to represent additional state but coordinating that with the PENDING bit got complex.